ion/skipper.go

```diff
@@ -804,6 +804,12 @@ func stopForCommentsHandler() (bool, error) {
 	return false, nil
 }
 
+// ensureNoCommentsHandler is a commentHandler that returns an
+// error if any comments are found, else no error is returned.
+func (t *tokenizer) ensureNoCommentsHandler() (bool, error) {
+	return false, &UnexpectedTokenError{"No Comments are allowed", t.Pos() - 1}
```

Comment:
This error message might be confusing without some additional context:

```suggestion
	return false, &UnexpectedTokenError{"Comments are not allowed within long-form strings.", t.Pos() - 1}
```

Reply:
Isn't it the case that we _can have_ comments in long-form string?
```
( '''hello '''     // Sexp with one element
  '''world!'''  )
```
And the restriction is only on clob, I assume.
How about "Comments are not allowed within text format clob." ?

Reply:
(Extreme nitpick: per https://github.com/golang/go/wiki/CodeReviewComments#error-strings error strings in go should not start with a capitalized word or end in punctuation.)

Reply:
> Isn't it the case that we _can have_ comments in long-form string?
> 
> ```
> ( '''hello '''     // Sexp with one element
>   '''world!'''  )
> ```
> 
> And the restriction is only on clob, I assume.
> How about "Comments are not allowed within text format clob." ?

Yes, that's correct. Since the binary format doesn't have comments, I think you can simplify it further to `comments are not allowed within a clob`.